tfe/provider.go

```diff
@@ -183,7 +183,7 @@ func getClient(tfeHost, token string, insecure bool) (*tfe.Client, error) {
 	// Make sure the transport has a TLS config.
 	transport := httpClient.Transport.(*http.Transport)
 	if transport.TLSClientConfig == nil {
-		transport.TLSClientConfig = &tls.Config{}
+		transport.TLSClientConfig = &tls.Config{MinVersion: tls.VersionTLS13}
```

Comment:
Do you happen to know what TLS versions are in use in TFC/older versions of TFE? How did you decide on this min version?

Reply:
By not specifying a `MinVersion`, it was defaulting to a version that created a security vulnerability. This min version was the latest per the docs.

Reply:
I do not know what TLS versions are in use in TFC/older versions of TFE, but I can look into it.

Reply:
🛑 The issue is that we could require a version greater than what versions of TFE would be capable of. We should either stick with the default (12) and create a linter exception or verify that version 13 is compatible with TFE versions going way back

Reply:
Thanks for raising this. I did some additional reading about the two versions and `TLS12` remains in widespread use.